### DISCUSSION TITLE
Is the try_runtime_upgrade error type needed?

### DISCUSSION THREAD

**Discussion by Chralt98** [paritytech/substrate#11504] [General]
When using `unwrap` it stops executing the code. Why is the return parameter [`&'static str` needed here](https://github.com/paritytech/substrate/blob/master/frame/executive/src/lib.rs#L265) then?

Wouldn't it have the same effect to replace it with the following?

```rust
pub fn try_runtime_upgrade() -> Result<frame_support::weights::Weight, ()>
```

Related to [this](https://github.com/KILTprotocol/kilt-node/issues/362).

**Comment by bkchr** [comment]
I don't get the question?

**Reply by Chralt98** [reply]
Sorry about that. Just for clarification: I thought, that `unwrap` would lead to stop the running code. Therefore I thought, that returning a `&' static str` would not have an effect to the caller of the `try_runtime_upgrade()` function. 

The second point when using this function: 
I saw the example of [KILT spiritnet here](https://github.com/KILTprotocol/kilt-node/blob/develop/runtimes/spiritnet/src/lib.rs#L1216) and wondered, if this `log::info` would ever be called. I reproduced the scenario [here](https://github.com/KILTprotocol/kilt-node/issues/362) and think still, that the `log::info` is unreachable there.

**Comment by shawntabrizi** [comment]
Copying the same thing I posted in your linked issue:

> You shouldn't be using `unwrap`, you should be handling the result with something like `?` or similar.
> 
> Of course if you `unwrap` and cause a panic, you will not trigger proper Rust error handling.

So, just like regular Rust, if you want to take advantage of Rust error handling, you should not be using `unwrap` or other panic inducing code.

On the other hand, if you correctly handle the Rust errors, you will see the `&'static str` printed for you, so this is why it is part of the signature of the function return.
